### ISSUE TITLE
Some problems in the concepts

### ISSUE THREAD

**Issue by jedieudonne** [cirosantilli/write-free-science-books-to-get-famous-website#3]
Hi,
Your algorithm to rank "teacher" is interresting but it has some drawbacks.

1/ **This algorithm is perfect for single questions answering.** On stackoverflow, it's perfect. You give a good and wel explained answer, you get up ranked. And that's all. In this case, it's easy to know if the "teacher" was good : He answered the question or he didn't. 
But for a course, it's much harder. When I teach computer science my first course is about the print function. After one hour the student only knows how to write "hello world". So I'm a bad teacher. But in the course I talked about types of objects, which will be very usefull.... some lessons later.... And all my students say that this trick is part of my expertise in computer science teaching. But they say that, because they don't judge only after one lesson. 

2/ **People will rank what they like, not what is the most usefull for them**.

- Making  a very easy course will cause people feel great, and will cause great ranking. 

- Making  a more complete but harder course will cause people make efforts, and may be fail to succeed, and then bad ranking.

**In conclusion** : If a teacher wants to be well ranked, he has to make people feel good, which is different of making people clever. I made you believe you intelligent so I'm well ranked. I asked you to make efforts to understand hard notions, I'm badly ranked.

**My suggestion** : keep you algorithm, he's usefull. But create a "expert council" to judge of the real pedagogical quality of a course. With those two indicators (expert ranking and user ranking) you'll get a very precise view of the course.

**Comment by cirosantilli** [comment]
Thanks for the feedback,

Yes, as you mention, those problems are too subjective to be solvable by machines, so I don't think much about them :-)

To me the main question is: is there an algorithm which will improve things, and which one is it.

Any algorithm can only be an approximation of "true value". PageRank managed to approximate "true value" well enough. I wonder if we can improve things significantly. Guess I'll only know when I try this out.

**Comment by jedieudonne** [comment]
I think your algorithm is great, but you can improve it with just this little addition 

In parallel with user ranking, just include a expert ranking. This expert ranking would be delivered by people who can justify a real expertise in the domain covered by the course. 

This way, you'll have two indications, user ranking will tell you how nice and how satifying a course is, and expert ranking will tell you how serious is the teacher, and how strong is the content. 
 
And you don't have to modify your algorithm, you just have to apply it to two differents populations : users and experts.

**Comment by cirosantilli** [comment]
@jedieudonne OK, I'll keep that in mind.
